frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerThread.java

```diff
@@ -120,12 +120,25 @@ public String getGpuUsage() {
             try {
                 // TODO : add a generic code to capture gpu details for different devices instead of
                 // just NVIDIA
-                process =
-                        Runtime.getRuntime()
-                                .exec(
-                                        "nvidia-smi -i "
-                                                + gpuId
-                                                + " --query-gpu=utilization.gpu,utilization.memory,memory.used --format=csv");
+                // Ensure gpuId is a valid number
+                int validatedGpuId;
+                try {
+                    validatedGpuId = Integer.parseInt(gpuId);
+                } catch (NumberFormatException e) {
+                    throw new IllegalArgumentException("Invalid GPU ID");
+                }
```

Comment:
GPU id is already validated during assignment stage.
- https://github.com/pytorch/serve/blob/master/frontend/archive/src/main/java/org/pytorch/serve/archive/model/ModelConfig.java#L236
- https://github.com/pytorch/serve/blob/master/frontend/server/src/main/java/org/pytorch/serve/wlm/WorkLoadManager.java#L210

Reply:
reverted gpu id validation but kep the process builder for gpu